### DISCUSSION TITLE
Print missing side walls

### DISCUSSION THREAD

**Discussion by nameauser** [FormerLurker/ArcWelderPlugin#148] [Q&A] [answered]
I used default settings in Octoprint and auto-generated a .aw.gcode. I can see the walls in the GcodeViewer tab in Octoprint, but those side wall lines/arcs are not coming out in the print, only bottom layers and infills . I am using Marlin 2.x bugfix on a SKR v1.3 2209 controller. I have also enabled G90/G91 in Features.

**Comment by FormerLurker** [comment, marked as answer]
Most likely arcs are not enabled on your printer.  Try sending a G2 command without any parameters.  If your printer responds with 'Unknown Command', arcs are disabled.  If they are disabled, you will need to either compile the firmware with arcs enabled, or find an existing binary that has arcs enabled.  Let me know how it goes!

**Reply by nameauser** [reply]
Thank you so much for your reply. In the _adv.h file that G2/G3 Arc Support was commented out exactly as you suspected. I have uncommented the main #define ARC_SUPPORT. There are a 5 other subordinate lines below which are still disabled, but I have not yet found any documentation on them to figure out if I could leave them be or make changes. In the Arc-Welder settings in Octoprint, I see resolution and maximum arc radius setting. So, do I need to enable  #define ARC_SEGMENT_PER_R for example and change #define MM_PER_ARC_SEGMENT  0.05? 

#define ARC_SUPPORT               // Disable this feature to save ~3226 bytes
#if ENABLED(ARC_SUPPORT)
  #define MM_PER_ARC_SEGMENT      1 // (mm) Length (or minimum length) of each arc segment
  //#define ARC_SEGMENTS_PER_R    1 // Max segment length, MM_PER = Min
  #define MIN_ARC_SEGMENTS       24 // Minimum number of segments in a complete circle
  //#define ARC_SEGMENTS_PER_SEC 50 // Use feedrate to choose segment length (with MM_PER_ARC_SEGMENT as the minimum)
  #define N_ARC_CORRECTION       25 // Number of interpolated segments between corrections
  //#define ARC_P_CIRCLES           // Enable the 'P' parameter to specify complete circles
  //#define CNC_WORKSPACE_PLANES    // Allow G2/G3 to operate in XY, ZX, or YZ planes
  //#define SF_ARC_FIX              // Enable only if using SkeinForge with "Arc Point" fillet procedure
#endif

**Reply by FormerLurker** [reply]
I wouldn't touch ANY of those defaults.  Best to see how they work (spoiler:  they work pretty darn well!) before tweaking anything.

> #define MM_PER_ARC_SEGMENT 0.05

That would probably blow your printer up, lol!  Well, not really, but it would slow it down to a crawl potentially.  So, that setting really determines the MINIMUM arc segment length.  If you are printing something with a reasonably diameter, it will look great.  

The other setting that is enabled by default is MIN_ARC_SEGMENTS.  This is sort of like 'facets' in scad.  It determines the minimum number of segments in a circle of the same radius as the G2/G3 you are sending.  That allows you to print tiny arcs that are smaller in radius than you mm_per_arc_segment setting, and they will always have at lease this number of segments.  However, Marlin has no min_mm_per_arc_segment setting, so this can cause a lot of interpolated segments that are unnecessary, but it's a good trade off at the moment.

Once you see what the defaults do, I would consider reducing mm_per_arc_segment to 0.5 and see if your board can keep up.  If it can't, try 0.75, etc..  You might get a SLIGHTLY better surface finish with a slightly lower setting, but it's hard to tell  0.5mm is not much at all :)

FYI, I'm trying to get a gcode added to control these settings (as well as a couple new ones).  This is a big missing feature IMHO, and I'd love to see it.  I've submitted a [PR for Prusa's MK2/MK3 firmware](https://github.com/prusa3d/Prusa-Firmware/pull/2657), since I have one to test on, and am building a printer now so i can run Marlin 2 and test my code.  Fingers crossed.

Good luck, and post back with any more questions!

**Reply by nameauser** [reply]
Thank you so much! Sorry I still know so little about Marlin - never even realised that there was a G2/G3 section. I did look at other Creality printers' _adv.h settings and could see that the same lines were also disabled, but MIN_ARC_SEGMENTS 16 instead of 24. I will leave everything alone, compile, upload and do a test run. Hopefully with great outcome and will report back regardless. Thank you again for your quick and thorough help.

**Comment by nameauser** [comment]
After enabling G2/G3 in _adv.h and flashed firmware.bin to the board, issuing G2 or G3 via Octoprint terminal actuall shut down the printer with a M112 accompanied by a message on Octoprint: Your printer's firmware reported an error. Due to that OctoPrint will disconnect. Reported error: G2/G3 bad parameters

**Reply by FormerLurker** [reply]
Yes, that's exactly what you should see.  Looks like you are good to go now!  The printer won't disconnect if you send a valid G2/G3.

**Reply by nameauser** [reply]
Great, thank you!

**Reply by FormerLurker** [reply]
Now that everything is in order, perhaps you would be willing to give the latest maintenance release candidate a go?  See [this guide](https://github.com/FormerLurker/ArcWelderPlugin/discussions/141) and if you decide to try it out, please leave feedback [here](https://github.com/FormerLurker/ArcWelderPlugin/issues/145).  I need more testers before I can get this working in the plugin manger.

**Comment by nameauser** [comment]
I will give the maintenance release a go. The stable release worked really well! Thank you for all your help and for creating this plugin.
